Docs: document report template variables (#31637)
grafana/grafana: docs/sources/enterprise/reporting.md

If the time zone is set differently between your Grafana server and its remote image renderer, then the time ranges in the report might be different between the page header and the time axes in the panels. We advise always setting the time zone to UTC for dashboards when using a remote renderer to avoid this.

> ## Report template variables

Reviewer: ```suggestion
## Report-template variables
```
Rationale: Disambiguate grammar where report could otherwise be a verb.

Author: Should it be smth like `Report-specific template variables` maybe? `Report-template` sound pretty confusing, at least to me.

Reviewer: How about `Variables of report templates`?

Author: `Template` here refers to dashboard templating, not report and in our docs we call [a query with a variable](https://grafana.com/docs/grafana/latest/variables/#templates) a template. So basically the functionality is about allowing users to select template variables for a dashboard from report.

> > Only available in Grafana Enterprise v7.5+ (behind `templateVariables` feature flag).

Author: `> Only available in Grafana Enterprise v7.x+` is used throughout the documentation (at least in enterprise docs), so changing it in one place will make the docs inconsistent.

> > Only available in Grafana Enterprise v7.5+ (behind `templateVariables` feature flag).

Reviewer: https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#notes-tips-cautions-and-warnings

> You can configure report-specific template variables for the dashboard on the report page. The selected variables will override the variables from the dashboard and will be used when rendering the report's PDF file. More detailed information about using template variables is available in the [Templates and variables]({{< relref "../variables/_index.md" >}}) section. 

Reviewer: ```suggestion
You can configure report-specific template variables for the dashboard on the report page. The variables that you select override the variables from the dashboard, and they are used when rendering a PDF file of the report. For detailed information about using template variables, refer to the [Templates and variables]({{< relref "../variables/_index.md" >}}) section. 
```

Reviewer: Rationale:
* Change passive voice to active voice.
* Change future to present tense.
* Conform to style guide wrt "..., refer to ...".
